x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/task/ReindexDataStreamStatus.java

```diff
@@ -69,12 +71,12 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
-        builder.field("start_time", persistentTaskStartTime);
+        builder.timestampFieldsFromUnixEpochMillis("start_time_millis", "start_time", persistentTaskStartTime);
         builder.field("complete", complete);
-        builder.field("total_indices", totalIndices);
+        builder.field("total_indices_in_data_stream", totalIndices);
         builder.field("total_indices_requiring_upgrade", totalIndicesToBeUpgraded);
-        builder.field("successes", totalIndicesToBeUpgraded - (inProgress + pending + errors.size()));
-        builder.field("in_progress", inProgress);
+        builder.field("successes", totalIndicesToBeUpgraded - (inProgress.size() + pending + errors.size()));
+        builder.field("in_progress", inProgress.size());
```

Comment:
Could inProgress by changed concurrently in another thread? If so, we could store `inProgress.size()` in a temp var to be sure that `successes` and `in_progress` are consistent

Reply:
Hmm yeah that could happen. I'm not really making any guarantees that these numbers reconcile (I think I'd have to synchronize access to this whole method to do that). But I can at least get rid of this particular race condition.